src/components/gltf-model-plus.js

```diff
@@ -341,127 +340,221 @@ function runMigration(version, json) {
+    gltf.scene.traverse(object => {
+      // GLTFLoader sets matrixAutoUpdate on animated objects, we want to keep the defaults
+      // @TODO: Should this be fixed in the gltf loader?
+      object.matrixAutoUpdate = THREE.Object3D.DefaultMatrixAutoUpdate;
+      const materialQuality = window.APP.store.materialQualitySetting;
+      object.material = mapMaterials(object, material => convertStandardMaterial(material, materialQuality));
+    });
 
-  const nodes = parser.json.nodes;
-  if (nodes) {
-    for (let i = 0; i < nodes.length; i++) {
-      const node = nodes[i];
+    //
+    gltf.scene.traverse(object => {
+      if (!object.material) {
+        return;
+      }
+      const materials = Array.isArray(object.material) ? object.material : [object.material];
+      // @TODO: Handle more efficiently
+      for (const material of materials) {
+        for (const key in material) {
+          const prop = material[key];
+          if (prop && prop.isTexture && !prop.onUpdate) {
+            prop.onUpdate = () => {
+              // Delete texture data once it has been uploaded to the GPU
+              prop.image.close && prop.image.close();
+              delete prop.image;
+            };
+          }
+        }
+      }
+    });
 
-      if (!node.extras) {
-        node.extras = {};
+    // Replace animation target node name with the node uuid.
+    // I assume track name is 'nodename.property'.
+    if (gltf.animations) {
+      for (const animation of gltf.animations) {
+        for (const track of animation.tracks) {
+          const parsedPath = THREE.PropertyBinding.parseTrackName(track.name);
+          for (const scene of gltf.scenes) {
+            const node = THREE.PropertyBinding.findNode(scene, parsedPath.nodeName);
+            if (node) {
+              track.name = track.name.replace(/^[^.]*\./, node.uuid + ".");
+              break;
+            }
+          }
+        }
       }
+    }
 
-      node.extras.gltfIndex = i;
+    //
+    if (this.fileMap) {
+      // The GLTF is now cached as a THREE object, we can get rid of the original blobs
+      Object.keys(this.fileMap).forEach(URL.revokeObjectURL);
     }
+
+    //
+    gltf.scene.animations = gltf.animations;
   }
+}
 
-  // Mark the special nodes/meshes in json for efficient parse, all json manipulation should happen before this point
-  parser.markDefs();
+class GLTFHubsLightMapExtension {
+  constructor(parser) {
+    this.parser = parser;
+    this.name = "MOZ_lightmap";
+  }
+
+  // @TODO: Ideally we should use extendMaterialParams hook.
+  //        But the current official glTF loader doesn't fire extendMaterialParams
+  //        hook for unlit and specular-glossiness materials.
+  //        So using loadMaterial hook as workaround so far.
+  //        Cons is loadMaterial hook is fired as _invokeOne so
+  //        if other plugins defining loadMaterial is registered
+  //        there is a chance that this light map extension handler isn't called.
```

Comment:
Since we have control over the loader this shouldnt really be an issue in practice, but we will need to be careful when doing upgrades that we don't accidentally break it by some plugin being introduced upstream. Hopefully this workaround doesn't last too long anyway.